File: unity-project/Assets/Scripts/BaseManager.js
#pragma strict

private var gameController : GameController;
private var count : int;
private var items : int;
private var levelDone : boolean;


function Start () {
	items = 0;
	levelDone = false;
	var gameControllerObject : GameObject = GameObject.FindWithTag("GameController");
	if(gameControllerObject != null){
		gameController = gameControllerObject.GetComponent(GameController);
	}
	
	if(gameControllerObject == null){
		Debug.Log("Can't fint 'GameController' script!");
	}
	var collectablesObject : GameObject = GameObject.FindWithTag("Collectables");
	
	if(collectablesObject != null)
	{
		count = collectablesObject.transform.childCount;
	}
}



function OnTriggerEnter(collision:Collider){
	if(collision.gameObject.tag == "Player"){
		items = count - gameController.GetCollected();
		Debug.Log(items);
		if(items == 0)
		{
			gameController.ChangeWinState();
			gameController.EndLevel();
		}
    }
}

function OnCollisionEnter(collision:Collision)
{
	Debug.Log("Ouch!");
	collision.gameObject.SendMessage("HitObject");
}

File: unity-project/Assets/Scripts/BaseManager.cs
using UnityEngine;
using System.Collections;

public class BaseManager : MonoBehaviour
{

	private GameController gameController;
	private int count;
	private int items;

	// Use this for initialization
	void Start ()
	{
		items = 0;
		gameController = GameObject.FindGameObjectWithTag("GameController").GetComponent<GameController>();
		GameObject collectablesObject = GameObject.FindWithTag ("Collectables");
		if(collectablesObject != null){
			count = collectablesObject.transform.childCount;
		}
	}
	
	// Update is called once per frame
	void Update ()
	{

	}

	void OnTriggerEnter(Collider collision) {
		if(collision.gameObject.tag == "Player"){
			gameController.RefillEnergy();
			items = count - gameController.GetCollected();
			if(items == 0)
			{
				gameController.ChangeWinState();
				gameController.EndLevel();
			}
		}
	}
	void OnCollisionEnter(Collision collision){
		collision.gameObject.SendMessage("HitObject");
	}
}

